test/ng/directive/asSpec.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+describe('as', function() {
+
+  it('should bind in the current scope the controller of a component', function() {
+    var expected = {some: 'value'};
+
+    angular.module('my', []).component('myComponent', {
+      template: 'foo',
+      controller: function() {
+        this.property = expected;
+      }
+    });
+    module('my');
+
+    inject(function($compile, $rootScope) {
+      var scope = $rootScope.$new();
+      var $ctrl = {undamaged: true};
+      scope.$ctrl = $ctrl;
+      dealoc($compile('<my-component as="$ctrl.myComponent"></my-component>')(scope));
+
+      expect(scope.$ctrl).toBe($ctrl);
+      expect(scope.$ctrl.undamaged).toBe(true);
+      expect($ctrl.myComponent).not.toBeUndefined();
+      expect($ctrl.myComponent && $ctrl.myComponent.property).toBe(expected);
+      scope.$destroy();
```

Comment:
No, I'm not confident, that's why I wrote _"(or dealoc the element, I believe)"_. TBH, when `dealoc` is needed and when it isn't is still a big mystery for me :).